### PR TITLE
Add esm support to emotion

### PR DESCRIPTION
**Pull request by TrySound**
Same as with rebase components.

- emotion.js
- emotion.mjs
- index.js
- index.mjs

`main` is referenced to dist/index.(js|mjs)
`/emotion` is referenced to dist/emotion.(js|mjs)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,14 +2,14 @@
   "name": "rebass",
   "version": "3.0.0-0",
   "description": "React primitive UI components built with styled-system",
-  "main": "dist/index.js",
-  "module": "dist/index.es.js",
```
Comment (jxnblk) [inline package.json]: Is the `module` field not required?

File: package.json [hunk 2]
```diff
@@ -2,14 +2,14 @@
...
+  "main": "dist/index",
   "sideEffects": false,
   "scripts": {
-    "prepare": "rm -rf dist && npm run build && npm run build:es && npm run build:emotion",
-    "build": "NODE_ENV=cjs babel src -d dist",
-    "build:es": "NODE_ENV=es babel src/index.js -o dist/index.es.js",
-    "build:emotion": "NODE_ENV=emotion babel src -o dist/emotion.js",
+    "prepare": "rm -rf dist && mkdir dist && npm run build:cjs && npm run build:esm && npm run build:emotion:cjs && npm run build:emotion:esm",
+    "build:cjs": "NODE_ENV=cjs babel src/index.js -o dist/index.js",
+    "build:esm": "NODE_ENV=esm babel src/index.js -o dist/index.mjs",
```
Comment (jxnblk) [inline package.json]: Does `.mjs` work with webpack and rollup? I thought that was only for Node.js

File: package.json (hunk 1 above)
Comment (TrySound) [reply package.json]: Nope with mjs

File: package.json (hunk 2 above)
Comment (TrySound) [reply package.json]: Both have priority to resolve mjs first
https://github.com/rollup/rollup-plugin-node-resolve/blob/master/src/index.js#L11
https://github.com/webpack/webpack/blob/master/lib/WebpackOptionsDefaulter.js#L327

Comment (jxnblk) [reply package.json]: That's certainly simpler, thanks!

**Comment by jxnblk** [comment]
This is looking good, thanks! Somewhat related to this, I'm trying to figure out the best way to replicate this babel setup for all the packages that will live in the `packages/` directory as well as have a single jest config for running tests for all the packages. Do you know if a root-level `babel.config.js`can easily be reused for each sub-package? Assuming that each package would be a single `index.js` file that should be exported with cjs, esm, and emotion versions

**Comment by TrySound** [comment]
Here's an example of root configs for all packages
https://github.com/smooth-code/svgr

I think build script will require some logic to iterate through each package. But it will be quite small.

**Comment by jxnblk** [comment]
Ah, looks like that just uses the `--config-file` flag to point to the root one then, right? Considering setting up Lerna, so I'm guessing that would handle what you mean with iterating through each package

**Comment by TrySound** [comment]
If you would like to omit package-lock.json in every sub package I recommend to go with `yarn` which has workspaces feature (simplifies lerna work).

**Comment by jxnblk** [comment]
Thanks for the suggestion, but yeah probably not gonna use yarn

**Pull request merged**